### ISSUE TITLE
Route Groups Not Working in v13.4.11

### ISSUE THREAD

**Issue by alwalxed** [vercel/next.js#52996]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #47~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 21 15:35:31 UTC 2
    Binaries:
      Node: 18.16.1
      npm: 9.5.1
      Yarn: 1.22.19
      pnpm: 8.6.9
    Relevant Packages:
      next: 13.4.11
      eslint-config-next: 13.4.11
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router

### Link to the code that reproduces this issue or a replay of the bug

Private Project

### To Reproduce

create a route group `app/(group)/section/page.tsx`

### Describe the Bug

![localhost-3000-search](https://github.com/vercel/next.js/assets/119763173/9ee35d26-c318-4826-8f56-1af4bde15530)

### Expected Behavior

ignoring the grouping route

**Comment by douglasduteil** [comment]
Since 13.4.10 too on my side. Revert to `next@13.4.9` temporally solves the issue

**Comment by ss098** [comment]
This error occurs when using Turbopack and Router Groups, which occurs on next@13.4.12.

**Comment by JoshApp** [comment]
I think this is fixed now in 13.5
